dataset/dataset_operations_common.h

```diff
@@ -117,6 +117,28 @@ DataArray apply_to_data_and_drop_dim(const DataArrayConstView &a, Func func,
                                       std::forward<Args>(args)...);
 }
 
+/// Helper for creating operations that return an object with modified data with
+/// all dimensions collapsed.
+///
+/// Examples are mostly reduction operations such as `sum` (dropping all
+/// dimensions at once). Creates new data
+/// array by applying `func` to data and dropping coords/masks/attrs depending
+/// The exception are multi-dimensional coords that depend on `dim`,
+/// with two cases: (1) If the coord is a coord for `dim`, `func` is applied to
+/// it, (2) if the coords is a coords for a dimension other than `dim`, a
+/// CoordMismatchError is thrown.
+///
+template <class Func>
+DataArray apply_to_data_and_drop_all_dims(const DataArrayConstView &a,
+                                          Func func) {
+  const auto &dimensions = a.data().dims();
+  DataArray out(a);
+  for (scipp::index i = 0; i < dimensions.ndim(); ++i) {
+    out = apply_to_data_and_drop_dim(out, func, dimensions.label(i), a.masks());
+  }
+  return out;
+}
+
```

Comment:
Why is this required? Is it not possible to use `reduce_all_dims`?

Reply:
Yes the logic is almost identical and can be templated. This does not however handle masks https://github.com/scipp/scipp/blob/master/variable/operations_common.h#L30-L38

Reply:
You don't need that? See how it is used: https://github.com/scipp/scipp/blob/53ed25a4778f6b5343b0dd177681b4803eefdae9/variable/reduction.cpp#L159-L162

Since `sum(data_array, dim)` exists you should be able to use the same. The masks are part of `data_array` and do not need special handling, I belive?